ingest/src/sources/mongodb-university/MongoDbUniversityDataSource.ts

```diff
@@ -0,0 +1,94 @@
+import { Page, PageMetadata } from "mongodb-rag-core";
+import { DataSource } from "../DataSource";
+import { makeUniversityPages } from "./makeUniversityPages";
+import {
+  TiCatalogItem,
+  makeMongoDbUniversityDataApiClient,
+} from "./MongoDbUniversityDataApiClient";
+
+/**
+  Parameters for constructing a MongoDB University Data API source.
+ */
+export interface MakeMongoDbUniversityDataSourceParams {
+  /**
+      Name of the MongoDB University Data API source.
+      Maps to the {@link DataSource.name}.
+     */
+  sourceName: string;
+
+  /**
+      Base URL for the MongoDB University Data API.
+     */
+  baseUrl: string;
+
+  /**
+      API key for the MongoDB University Data API.
+     */
+  apiKey: string;
+
+  /**
+      Filter function for filtering out items from the MongoDB University
+      catalog. For example, you may want to only ingest items that are
+      in public content.
+
+      To only ingest public, published, and non-legacy
+      MongoDB University content, use the
+      {@link filterOnlyPublicActiveTiCatalogItems} filter function.
+     */
+  tiCatalogFilterFunc: (item: TiCatalogItem) => boolean;
```

Comment:
Separately from ingest/this work, it might be nice to have some type of stronger guarantee that our public cluster never contains or exposes private data. I'm thinking something like a db trigger or schema that ensures `item.microsites.includes("University")` etc.

Reply:
IMO a stronger guarantee would be better at the University Data API level